src/main/groovy/io/seqera/wave/controller/ViewController.groovy

```diff
@@ -233,8 +233,22 @@ class ViewController {
         }
         //add conda lock file when available
         if( buildLogService && result.condaFile ) {
-            binding.build_conda_lock_data = buildLogService.fetchCondaLockString(result.buildId)
-            binding.build_conda_lock_url = "$serverUrl/v1alpha1/builds/${result.buildId}/condalock"
+            def condaLock = buildLogService.fetchCondaLockString(result.buildId)
+            def buildId = result.buildId
+
+            if ( condaLock && condaLock.contains('cat environment.lock')) {
+                condaLock = null //if there is no conda lock file, don't show the conda lock data
+                def builds = persistenceService.allBuilds(result.buildId.split('-')[1].split('_')[0])
+                for (def build : builds) {
+                    if (build.succeeded()){
+                        buildId = build.buildId
+                        condaLock = buildLogService.fetchCondaLockString(build.buildId)
+                    }
+                }
+            }
+
+            binding.build_conda_lock_data = condaLock
+            binding.build_conda_lock_url = "$serverUrl/v1alpha1/builds/${buildId}/condalock"
```

Comment:
the problem is when an image is cached, the dockerfile does not get executed and in that case condalock will contain `cat environment.lock` because its not been executed, so we will check the previous successful build and render that instead of the current one 
We can disable caching to avoid this situation

Reply:
Ok, let's add an extensive comment like the one above to document this need

Reply:
done

Reply:
added break to not check all the builds